copystructure.go

```diff
@@ -2,27 +2,19 @@ package copystructure
 
 import (
 	"reflect"
+	"sync"
 
 	"github.com/mitchellh/reflectwalk"
 )
 
 // Copy returns a deep copy of v.
 func Copy(v interface{}) (interface{}, error) {
-	w := new(walker)
-	err := reflectwalk.Walk(v, w)
-	if err != nil {
-		return nil, err
-	}
-
-	// Get the result. If the result is nil, then we want to turn it
-	// into a typed nil if we can.
-	result := w.Result
-	if result == nil {
-		val := reflect.ValueOf(v)
-		result = reflect.Indirect(reflect.New(val.Type())).Interface()
-	}
+	return Config{}.Copy(v)
+}
 
-	return result, nil
+// LockedCopy returns a deep copy of v, taking locks as needed during the walk.
+func LockedCopy(v interface{}) (interface{}, error) {
```

Comment:
One more nitpick: I think we just remove this and if you want locks then you use the `Config` structure.